lib/game.rb

```diff
@@ -0,0 +1,75 @@
+class Game
+  def initialize(args)
```

Comment:
After reading your comment, I thought about how this could be broken up. I made two new classes called `GameState` and `GameMessenger`. 

In POODR, they mentioned that hashes are better as arguments because it's much more flexible to change. Having multiple arguments creates a dependency on argument order. My one concern with this approach is that Ruby throws an error if you pass in the wrong number of arguments. With a hash, I don't get errors when I'm not passing in all the expected arguments.

Reply:
Totally agreed on argument order and that hashes as argument containers have benefits. The main tradeoff is that you lose context when the dependencies are all placed under a single less meaningful name like `args` or `params`.

A nice middle ground you can look into is named args.
```
def initialize(game:, board:) {...}
```
These are not order dependent and are just syntactic sugar around a hash of arguments and you can even provide defaults! (though defaults can be seriously abused so be careful!)